mrp_bom_current_stock/wizard/bom_route_current_stock.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Camptocamp SA
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class BomRouteCurrentStock(models.TransientModel):
+    _name = "mrp.bom.current.stock"
+
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom",
+        string="Starting Bill of Materials",
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product Variant',
+        domain="[('type', 'in', ['product', 'consu'])]",
+        required=True,
+    )
+    product_tmpl_id = fields.Many2one(
+        comodel_name='product.template',
+        string='Product Template',
+        related='product_id.product_tmpl_id',
+    )
+    product_qty = fields.Float(
+        related='bom_id.product_qty',
+        digits=dp.get_precision('Product Unit of Measure'),
+    )
+    product_uom_id = fields.Many2one(
+        comodel_name="product.uom",
+        related="bom_id.product_uom_id",
+    )
+    location_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Starting location",
+    )
+    line_ids = fields.One2many(
+        comodel_name='mrp.bom.current.stock.line',
+        inverse_name='explosion_id',
+    )
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.bom_id = self.env['mrp.bom']._bom_find(
+            product_tmpl=self.product_id,
+        )
+
+    @api.onchange('bom_id')
+    def _onchange_bom_id(self):
+        if self.bom_id.location_id:
+            self.location_id = self.bom_id.location_id
+
+    @api.model
+    def _prepare_line(self, bom_line, level, factor):
+        return {
+            'product_id': bom_line.product_id.id,
+            'bom_line': bom_line.id,
+            'bom_level': level,
+            'product_qty': bom_line.product_qty * factor,
+            'product_uom_id': bom_line.product_uom.id,
+            'location_id': (
+                bom_line.location_id.id if bom_line.location_id else False,
+            ),
```

Comment:
This is interpreted as a tuple with False inside, breaking the create on line 79.
```
> /opt/odoo/external-src/manufacture-reporting/mrp_bom_current_stock/wizard/bom_route_current_stock.py(80)_create_lines()
-> line_obj.create(vals)
(Pdb) vals
{'bom_line': 51, 'product_qty': 200.0, 'location_id': (False,), 'product_id': 95, 'bom_level': 1, 'explosion_id': 1, 'product_uom_id': 11}
```